### PR TITLE
feat(Character.AI): add presence

### PR DESCRIPTION
**Pull request by Faris0520**
## Description 
Added new presences for [Character.AI](https://beta.character.ai)

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image (1)

image (2)

image

</details>

### CODE REVIEW

File: websites/C/Character.AI/presence.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+const presence = new Presence({
+    clientId: "1061324603022114998",
+}),
+browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", () => {
+const presenceData: PresenceData = {
+    largeImageKey: "logo",
```
Comment (DarkVillager) [inline websites/C/Character.AI/presence.ts:8]: We're changing all largeImageKeys to imgur links
```suggestion
    largeImageKey: "https://i.imgur.com/eIpvMGf.png",

```

File: websites/C/Character.AI/presence.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    presenceData.details = "Browsing chats";
+
+} else if (document.location.pathname.startsWith("/community")) {
+    presenceData.details = "Viewing the community tab";
+
+} else if (document.location.pathname.startsWith("/profile")) {
+    presenceData.details = "Viewing my profile";
+    
+}
+else presenceData.details = "Browsing...";
```
Comment (DarkVillager) [inline websites/C/Character.AI/presence.ts:54]: Change this to a switch like here (make sure to also copy the variables that are shown here):
https://github.com/PreMiD/Presences/blob/fa35cfec5122d49de396493b36b48cbfbf5e34b2/websites/K/Kinopub/presence.ts#L69-L86

**Review by DarkVillager — COMMENTED** [review]
Also include links like [these](https://beta.character.ai/posts?topic=SGYIzinY5rZgKxsy8n2esQWg3_q5L7MnAF1VMh-YLas) with in this case Announcements:

File: websites/C/Character.AI/metadata.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+		"name": "parr1s"
+	},
+	"service": "Character.AI",
+	"description": {
+		"en": "Character.AI lets you create Characters and talk to them. Intelligent agents live here!",
+		"id": "Character.AI memungkinkan Anda membuat Karakter dan berbicara dengan mereka. Kecerdasan buatan disini!"
+	},
+	"url": [
+		"beta.character.ai"
+	],
```
Comment (DarkVillager) [inline websites/C/Character.AI/metadata.json:14]: [] are for multiple urls so remove them

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Review by Faris0520 — COMMENTED** [review]
edited, 👍

File: websites/C/Character.AI/presence.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            break;
+
+        case "chat":
+            presenceData.details = "Chatting with";
+
+            let char = document
+            .querySelector("head > title")
+            .textContent.replace("Character.AI - ", "");
+            
+            let img = document
```
Comment (Bas950) [inline websites/C/Character.AI/presence.ts:25]: combine `let`s

**Review by Bas950 — CHANGES_REQUESTED** [review]
run prettier

File: websites/C/Character.AI/presence.ts (hunk 4 above)
Comment (Faris0520) [reply websites/C/Character.AI/presence.ts:25]: what?

Comment (Faris0520) [reply websites/C/Character.AI/presence.ts:25]: what?

Comment (DarkVillager) [inline websites/C/Character.AI/presence.ts:25]:
```suggestion
            .textContent.replace("Character.AI - ", ""),
            img = document
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Review by Faris0520 — COMMENTED** [review]
edited

**Comment by Faris0520** [comment]
I don't know what's wrong with code. I run prettier 3 times and still error

image

_sorry for that, im new for this_ 🙏

File: websites/C/Character.AI/presence.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+			break;
+
+		case "profile":
+			presenceData.details = "Viewing my profile";
+			break;
+	}
+
+	if (presenceData.details) presence.setActivity(presenceData);
+	else presence.setActivity();
+});
```
Comment (DarkVillager) [inline websites/C/Character.AI/presence.ts:85]: This should fix the prettier issue and the requested changes. Next time use Visual Studio Code cuz that way prettier works :) 
```suggestion
const presence = new Presence({
		clientId: '1061324603022114998',
	}),
	browsingTimestamp = Math.floor(Date.now() / 1000);

presence.on('UpdateData', () => {
	const presenceData: PresenceData = {
			largeImageKey: 'https://i.imgur.com/eIpvMGf.png',
			startTimestamp: browsingTimestamp,
		},
		{ pathname } = document.location;

	switch (pathname.split('/')[1]) {
		case '':
			presenceData.details = 'In the Homepage';
			break;

		case 'chat':
			presenceData.details = 'Chatting with';

			let char = document
					.querySelector('head > title')
					.textContent.replace('Character.AI - ', ''),
				img = document
					.querySelector("meta[property='og:image']")
					.getAttribute('content');

			presenceData.largeImageKey = `${img.replace('80', '400')}`;
			presenceData.state = `${char}`;
			presenceData.buttons = [
				{ label: `Chat ${char}`, url: document.location.href },
			];
			break;

		case 'feed':
			presenceData.details = 'Browsing the feed';
			break;

		case 'post':
			let title3 = document.querySelector('head > title').textContent;

			presenceData.details = 'Viewing a post';
			presenceData.state = `${title3}`;
			break;

		case 'posts':
			let title = document.querySelector(
					'.d-flex.flex-row.align-items-center'
				).textContent,
				pict = document.querySelector('.sb-avatar__image').getAttribute('src');

			presenceData.details = 'Browsing posts';
			presenceData.state = `\"${title}\"`;
			presenceData.smallImageKey = `${pict.replace('80', '400')}`;
			break;

		case 'signup':
			presenceData.details = 'Signing up';
			break;

		case 'character':
			if (pathname.split('/')[2] === 'create') {
				presenceData.details = 'Creating a character';
			}
			presenceData.details = 'Creating a character';
			break;

		case 'chats':
			presenceData.details = 'Browsing chats';
			break;

		case 'community':
			presenceData.details = 'Viewing the community tab';
			break;

		case 'profile':
			presenceData.details = 'Viewing my profile';
			break;
	}

	if (presenceData.details) presence.setActivity(presenceData);
	else presence.setActivity();
});

```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/C/Character.AI/presence.ts (hunk 5 above)
Comment (Faris0520) [reply websites/C/Character.AI/presence.ts]: hmm, still not work 🤔

**Comment by Faris0520** [comment]
uhhhh

**Comment by Faris0520** [comment]
I've done

**Comment by EncryptedDev** [comment]
Hello there, 

Unfortunately, I am not satisfied with the quality of this pull request, therefore, will be closing it. 
- Unnecessary comments 
- Low quality presence - generally doesn't offer a range of features (including static **and** dynamic features) 
- Not needed ESLint rules disabled

**Pull request closed without merging**